synedrion/src/tools/bitvec.rs

```diff
@@ -17,6 +17,20 @@ impl BitVec {
         rng.fill_bytes(&mut bytes);
         Self(bytes.into())
     }
+
+    pub fn xor_all<'a>(bitvecs: impl Iterator<Item = &'a Self>) -> Self {
+        let mut bitvecs = bitvecs;
+        let mut result = bitvecs.next().unwrap().clone();
+        for bitvec in bitvecs {
+            result ^= bitvec;
+        }
+        result
```

Comment:
Would this work here?

```rust
        bitvecs.fold(Self::default(), |mut result, bitvec| {
            result ^= bitvec;
            result
        })
```

Reply:
Yes, that looks better. Maybe should even go into a `BitVec` method.